src/ledger/test/LedgerTxnTests.cpp

```diff
@@ -2804,6 +2819,308 @@ TEST_CASE("Erase performance benchmark", "[!hide][erasebench]")
+        std::vector<LedgerKey> deadKeyVect{deadKeySet.begin(),
+                                           deadKeySet.end()};
+
+        app->getBucketManager().addBatch(*app, 2, 20, {}, ledgerVect,
+                                         deadKeyVect);
+        ltx.commit();
+
+        auto addTxn = [&](int32_t offset, std::vector<LedgerEntry> entries,
+                          std::vector<LedgerKey> deadKeys = {}) {
+            SorobanResources resources;
+
+            for (auto& e : entries)
+            {
+                resources.readBytes += xdr::xdr_size(e);
+                auto k = LedgerEntryKey(e);
+                keysToPrefetch.emplace(k);
+                // Randomly add the key to either the read or write set.
+                if (stellar::rand_flip())
+                {
+                    resources.footprint.readOnly.emplace_back(k);
+                }
+                else
+                {
+                    resources.footprint.readWrite.emplace_back(k);
+                }
+                // If the key is not a TTL and its offset is less than 0 and
+                // it has not been marked as expected to succeed in another
+                // transaction, add it to the expected failed keys set.
+                if (k.type() != TTL && offset < 0 &&
+                    expectedSuccessKeys.find(k) == expectedSuccessKeys.end())
+                {
+                    expectedFailedKeys.emplace(k);
+                }
+                else
+                {
+                    expectedSuccessKeys.emplace(k);
+                    // If a key has sufficient in one transaction, remove it
+                    // from the expected failed keys set.
+                    expectedFailedKeys.erase(k);
+                }
+            }
+            if (offset + resources.readBytes < 0)
+            {
+                resources.readBytes = 0;
+            }
+            else
+            {
+                resources.readBytes += offset;
+            }
+
+            for (auto& k : deadKeys)
+            {
+                resources.footprint.readOnly.emplace_back(k);
+            }
+
+            REQUIRE(resources.footprint.readOnly.size() +
+                        resources.footprint.readWrite.size() ==
+                    entries.size() + deadKeys.size());
+            lkMeterHot->addTxn(resources);
+            lkMeterCold->addTxn(resources);
+            ensureLedgerKeyMeterCanLoad(
+                lkMeterHot, entries, expectedSuccessKeys, expectedFailedKeys);
+            ensureLedgerKeyMeterCanLoad(
+                lkMeterCold, entries, expectedSuccessKeys, expectedFailedKeys);
+        };
+        auto checkPrefetch = [&](std::vector<LedgerEntry>& allEntries) {
+            LedgerTxn ltx2(root);
+            auto numLoadedCold =
+                root.prefetchSoroban(keysToPrefetch, lkMeterCold);
+            REQUIRE(numLoadedCold >= expectedSuccessKeys.size());
```

Comment:
Once we pass in an explicit `expectedSuccess` instead of generating one in `addTxn`, we can enforce this more tightly with `REQUIRE(numLoadedCold == expectedSuccessKeys.size());`